File: backend/app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Customer extends Model
{
    protected $fillable = [
        'name', 'contact_person', 'email', 'phone',
        'city', 'address', 'tax_no',
    ];

    public function orders()
    {
        return $this->hasMany(Order::class);
    }

    // Toplam sipariş değeri hesabı
    public function getOrderCountAttribute(): int
    {
        return $this->orders()->count();
    }
}

File: backend-dotnet/Models/Customer.cs
namespace TexTrack.Api.Models;

public class Customer
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public string? City { get; set; }
    public string? Address { get; set; }
    public string? TaxNo { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<Order> Orders { get; set; } = new List<Order>();
}
